pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,378 @@
+            // triggered an unmount.  Thus, we check synchronously
+            // whether or not we should be doing a offline resize or
+            // not.
+            //
+            // Another option for us would be to just mount the
+            // filesystem back if that's what we expect, to undo the
+            // bug mentioned above. But let's be a bit more passive
+            // here and hope the bug gets fixed eventually.
+            return (is_mounted_synch(client.blocks[fsys.path])
+                    .then(is_mounted => {
+                        // When doing an offline resize, we need to first repair the filesystem.
+                        if (!is_mounted) {
+                            return (fsys.Repair({ })
+                                    .then(function () { return fsys.Resize(size - crypto_overhead, { }) }));
+                        } else {
+                            return fsys.Resize(size - crypto_overhead, { });
+                        }
+                    }));
+        } else if (vdo) {
+            if (size - crypto_overhead > vdo.physical_size)
+                return vdo.grow_physical();
+            else if (size - crypto_overhead < vdo.physical_size)
+                return Promise.reject(_("VDO backing devices can not be made smaller"));
+            else
+                return Promise.resolve();
+        } else if (stratis_bdev) {
+            const delta = size - crypto_overhead - Number(stratis_bdev.TotalPhysicalSize);
+            if (delta > 0) {
+                const pool = client.stratis_pools[stratis_bdev.Pool];
+                return pool.GrowPhysicalDevice(stratis_bdev.Uuid).then(std_reply);
+            } else if (delta < 0)
+                return Promise.reject(_("Stratis blockdevs can not be made smaller"));
+            else
+                return Promise.resolve();
+        } else if (size < orig_size) {
+            // This shouldn't happen.  But if it does, continuing is harmful, so we throw an error.
+            return Promise.reject(_("Unrecognized data can not be made smaller here."));
+        } else {
+            // Growing unrecognized content, nothing to do.
+            return Promise.resolve();
+        }
+    }
+
+    function crypto_resize() {
+        if (crypto) {
+            const opts = { };
+            if (passphrase)
+                opts.passphrase = { t: "s", v: passphrase };
+            return crypto.Resize(size - crypto_overhead, opts);
+        } else {
+            return Promise.resolve();
+        }
+    }
+
+    function lvm_resize() {
+        if (size != lvol.Size)
+            return lvol.Resize(size, { });
+        else
+            return Promise.resolve();
+    }
+
+    if (fsys && !fsys.Resize) {
+        // Fallback for old versions of UDisks.  This doesn't handle encrypted volumes.
+        if (size != orig_size) {
+            return lvol.Resize(size, { resize_fsys: { t: 'b', v: true } });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19400-20230927-061737-0f5b7102-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Reply:
Unneeded fallback